lib/osf-components/addon/components/addons-service/user-addons-manager/component.ts

```diff
@@ -125,8 +141,8 @@ export default class UserAddonManagerComponent extends Component<Args> {
     @waitFor
     async getStorageAddonProviders() {
         const activeFilterObject = this.filterTypeMapper[FilterTypes.STORAGE];
-        const serviceStorageProviders: AllProviderTypes[] =
-            await taskFor(this.getExternalProviders).perform(activeFilterObject.modelName);
+        const serviceStorageProviders = await taskFor(this.getExternalProviders)
+            .perform(activeFilterObject.modelName) as ExternalStorageServiceModel[];
```

Comment:
Had to go back and figure out why this one is doing [type assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) (`... as ExternalStorageServiceModel[]`) instead of [type annotation](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-annotations-on-variables) (`const xyzProviders: XYZServiceModel[] = ...`), and it is because `ExternalStorageService` has some fields on it that don't exist in the other `CloudComputing/CitationServiceModel`, so TS will say "Type 'AllProviderTypes' is not assignable to type 'ExternalStorageServiceModel'" if we don't override it with type assertion. I think the other two tasks will eventually get a type error when we fill out the CloudComputingService and CitationService models, so I will update those to use type assertions as well.

Reply:
Also, this one uses an intermediate variable `sortedList` while the others don't. One of them sets the list as an EmberArray while the others don't... This is all over the place for a copy-paste job 😆